### PR TITLE
wait for response form - numeric rule - rule translations tab - has number between and empty translations bugfixes

### PR DESCRIPTION
**Pull request by susanm74**
https://github.com/nyaruka/floweditor/issues/1081

### CODE REVIEW

File: src/components/flow/routers/localization/RouterLocalizationForm.tsx [hunk 1]
```diff
@@ -119,7 +119,9 @@ export default class RouterLocalizationForm extends React.Component<
       const { verboseName } = getOperatorConfig(originalCase.type);
 
       if (originalCase.arguments) {
-        const [orginalArgument] = originalCase.arguments;
```
Comment (susanm74) [inline src/components/flow/routers/localization/RouterLocalizationForm.tsx:122]: i tried googling, but couldn't find any refs or examples of declaring a variable this way `const [orginalArgument]` - is this syntax some sort of var declaration shorthand?

**Comment by susanm74** [comment]
has number between bugfix
BEFORE
Screen Shot 2022-10-27 at 1 18 09 PM
AFTER
Screen Shot 2022-10-27 at 1 22 39 PM

**Comment by susanm74** [comment]
saving empty rule translations bugfix
BEFORE
Screen Shot 2022-10-27 at 1 20 22 PM

AFTER
Screen Shot 2022-10-27 at 1 24 14 PM

**Comment by susanm74** [comment]
@ericnewcomer @norkans7 @rowanseymour a gentle nudge here

**Comment by ericnewcomer** [comment]
What do we do in the rule localization? I thought it was the argument list that was the primary issue.

**Comment by rowanseymour** [comment]
I thought we said we were going to get rid of localizing numerical rule arguments...

**Comment by ericnewcomer** [comment]
Ya, my understanding is that the fix we really need is the filtering out of has_number_between rules from the list of localizable arguments.

**Comment by susanm74** [comment]
AFTER - with all "has number between" categories - rule translations tab hidden
Screen Shot 2022-11-02 at 2 42 59 PM
Screen Shot 2022-11-02 at 2 39 12 PM

AFTER - with a mix of "has number between" categories and other categories - rule translations tab displays but filters those out from the list
Screen Shot 2022-11-02 at 2 43 10 PM
Screen Shot 2022-11-02 at 2 39 23 PM
Screen Shot 2022-11-02 at 2 39 36 PM

**Comment by susanm74** [comment]
@ericnewcomer @norkans7 @rowanseymour RFR again, now also includes the fix for filtering out "has number between" rules from the rule translations list

**Comment by norkans7** [comment]
What about the other numerical rules types such as `Operators.has_number_lt`, `Operators.has_number_gt`?
I think their arguments as well should not be translated

**Comment by susanm74** [comment]
@norkans7 i think we're just targeting this numeric rule for now because it's the only numeric rule that involves 2 arguments (ranges with an lower and upper bound), all of the others only contain 1 argument, and thus, aren't causing the issues this 2 argument numeric rule was causing, but yeah, i do think if we wanted to exclude those as well, we could easily exclude those from translation in a follow-up PR

File: src/components/flow/routers/localization/RouterLocalizationForm.tsx [hunk 2]
```diff
@@ -194,7 +198,11 @@ export default class RouterLocalizationForm extends React.Component<
 
     const hasCasesWithArguments = !!this.state.cases.find((kase: Case) => {
       const orginalCase = getOriginalCase(this.props.nodeSettings, kase.uuid) as Case;
-      return orginalCase.arguments && orginalCase.arguments.length > 0;
+      return (
+        orginalCase.arguments &&
+        orginalCase.arguments.length > 0 &&
+        orginalCase.type !== Operators.has_number_between
```
Comment (ericnewcomer) [inline src/components/flow/routers/localization/RouterLocalizationForm.tsx:204]: This one-liner would be more robust:
`(originalCase.arguments || []).length == 1`

**Pull request merged**